README
NAME
    Pinto::Server - Web interface to a Pinto repository

VERSION
    version 0.50

ATTRIBUTES
  root
    The path to the root directory of your Pinto repository. The repository
    must already exist at this location. This attribute is required.

  auth
    The hashref of authentication options, if authentication is to be used
    within the server. One of the options must be 'backend', to specify
    which Authen::Simple:: class to use; the other key/value pairs will be
    passed as-is to the Authen::Simple class.

  router
    An object that does the Pinto::Server::Handler role. This object will do
    the work of processing the request and returning a response.

  default_port
    Returns the default port number that the server will listen on. This is
    a class attribute.

METHODS
  to_app()
    Returns the application as a subroutine reference.

  call( $env )
    Invokes the application with the specified environment. Returns a
    PSGI-compatible response.

    There is nothing to see here.

    Look at pintod if you want to start the server.

SUPPORT
  Perldoc
    You can find documentation for this module with the perldoc command.

      perldoc Pinto::Server

  Websites
    The following websites have more information about this module, and may
    be of help to you. As always, in addition to those websites please use
    your favorite search engine to discover more resources.

    *   Search CPAN

        The default CPAN search engine, useful to view POD in HTML format.

        <http://search.cpan.org/dist/Pinto-Server>

    *   CPAN Ratings

        The CPAN Ratings is a website that allows community ratings and
        reviews of Perl modules.

        <http://cpanratings.perl.org/d/Pinto-Server>

    *   CPAN Testers

        The CPAN Testers is a network of smokers who run automated tests on
        uploaded CPAN distributions.

        <http://www.cpantesters.org/distro/P/Pinto-Server>

    *   CPAN Testers Matrix

        The CPAN Testers Matrix is a website that provides a visual overview
        of the test results for a distribution on various Perls/platforms.

        <http://matrix.cpantesters.org/?dist=Pinto-Server>

    *   CPAN Testers Dependencies

        The CPAN Testers Dependencies is a website that shows a chart of the
        test results of all dependencies for a distribution.

        <http://deps.cpantesters.org/?module=Pinto::Server>

  Bugs / Feature Requests
    <https://github.com/thaljef/Pinto-Server/issues>

  Source Code
    <https://github.com/thaljef/Pinto-Server>

      git clone git://github.com/thaljef/Pinto-Server.git

AUTHOR
    Jeffrey Ryan Thalhammer <jeff@imaginative-software.com>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2011 by Imaginative Software Systems.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
0.50      2012-09-18 17:55:41 America/Los_Angeles

     Now requires Pinto-0.052, which stores the index for
     each stack in a static file.  This makes the server much
     simpler and faster.  But beware that repositories created
     with older versions of Pinto are not compatible.  Contact
     thaljef@cpan.org if you need a migration path.

0.049     2012-08-15 14:25:34 America/Los_Angeles

     Fixed the logging mechanism so clients can produce
     colorized log messages.

0.048     2012-08-14 16:18:08 America/Los_Angeles

     Was not serving the 03modlist.data.gz and 01mailrc.txt.gz
     files.  These are required by the old cpan[1] client.
     
0.047     2012-08-14 14:23:11 America/Los_Angeles

     Was not serving up the default 02packages.txt.gz file
     when it was requested without an explicit stack name
     in the request URL.

     Improved signal handling and tempfile cleanup.

0.046     2012-08-13 17:23:25 America/Los_Angeles

    Added a Pinto::Server::Tester which is useful for testing
    a repository using a live server.

0.042     2012-07-15 01:40:58 America/Los_Angeles

    Pinto::Server now supports stacks, and requires Pinto version
    0.044 or newer.  Clients must also use Pinto::Remote version 0.041
    or newer.

    The "pinto-server" command has been renamed to "pintod".

0.037     2012-04-09 12:57:50 America/Los_Angeles

    Now sends all log messages (at any level) to the client.  It
    is up to the client to decide which ones it wants to record
    or display.

0.036     2012-04-08 23:46:46 America/Los_Angeles

    If using Starman or Twiggy, Pinto will abort gracefully
    if the connection with the client is dropped (like if
    you press Ctrl-C when using pinto-remote).

    Now requires Pinto 0.036 which has a different interface
    for controlling where log messages go (Inspired by
    Karen Etheridge).

    Added Authen::Simple::Passwd as test prereq to fix build failures
    (Thanks CPAN Testers!)

0.035     2012-04-04 20:35:00 America/Los_Angeles

    New Features:

    pinto-server can now be configured to require authentication.
    Authentication can be done using any of the Authen::Simple
    modules.  See the POD for details.  If you use this feature, then
    you'll also need to use a version of pinto-remote that
    supports authentication.  (Karen Etheridge)

    pinto-server now streams back the output from Actions.  You will
    need to use a backend server that supports streaming for this to
    work (e.g. Starman, Twiggy, etc.)  Fortunatley, the default
    backend (HTTP::Server::PSGI) does support streaming.

    Other Stuff:

    Pinto::Server has been completely redesigned into a simple web
    application micro-framework.  This eliminates our dependency on
    Dancer (not that there is anything wrong with Dancer.  I just
    needed something a little less frameworky).

    In theory, Pinto::Server now scales to support any future Action
    without having to write any additional code here.  As long as your
    Action follows convention and behaves properly, then the server
    will just do-the-right-thing.

0.034     2012-03-17 08:47:01 America/Los_Angeles

    Bug Fixes:

    pinto-server had a broken shebang.

0.033     2012-03-17 00:59:47 America/Los_Angeles

    Support the --norecurse option in the add command (Karen
    Etheridge) which also requires Pinto-0.033 or newer.

0.029     2012-01-26 22:25:06 America/Los_Angeles

    The --repos option on pinto-server has been officially renamed to
    --root.  This is to be consistent with the interfaces of
    pinto-admin and pinto-remote.  The old --repos option will not be
    supported for backward compatibility.

0.028     2011-12-12 00:23:45 America/Los_Angeles

    Exceptions from Pinto are now trapped and returned in the body of
    the response from the server.  This is more helpful than the 500
    error page that Dancer throws up.

    Added support for the statistics, pin, and unpin actions.  So
    we now need Pinto version 0.028 or better.

0.026     2011-12-07 14:09:41 America/Los_Angeles

    No new functionality.  Just matching up with the latest Pinto API.
    Pinto and Pinto::Server are both evolving rapidly, and I make no
    promises about backward compatibility right now.

0.021     2011-09-03 21:43:25 America/Los_Angeles

    Addd the nop route.  This is used by the 'ping' command of
    pinto-remote.

0.020     2011-09-01 15:29:40 America/Los_Angeles

    Now requires Pinto 0.23, which has a different interface.

    Refactored internals so they could be tested.

    And then I actually wrote the tests!

0.019     2011-08-30 01:14:43 America/Los_Angeles

    Now requires Pinto 0.021, which behaves better in a server
    environment.  I thought I had fixed this in version 0.017, but I
    was wrong.

0.018     2011-08-28 20:59:15 America/Los_Angeles

    Now required Pinto 0.020, which supports the Nop action.

    pinto-server will now always pull/update your repository from VCS
    upon startup.  This helps ensure that the repository is in a good
    state when we start.  But we never pull/update again after that.

0.017     2011-08-24 00:24:17 America/Los_Angeles

    Now requires Pinto 0.017 or newer, which runs better under a
    server environment.

    All the guts of Pinto::Server are now in pinto-server, as this
    seemed to be the only way to integrate my Dancer routes with my
    own command-line switches without having to use Dancer's own
    configuration API.  I'm quite sure that I'm not using Dancer the
    way it is intended.

0.001     2011-08-17 01:04:58 America/Los_Angeles

    Initial release.


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Apache::Htpasswd" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Find" => 0,
    "File::Spec::Functions" => 0,
    "HTTP::Request" => 0,
    "HTTP::Request::Common" => 0,
    "List::Util" => 0,
    "Module::Build" => "0.3601",
    "PerlIO::gzip" => 0,
    "Pinto::Tester::Util" => 0,
    "Plack::Test" => 0,
    "Test::More" => 0
  },
  "configure_requires" => {
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "Web interface to a Pinto repository",
  "dist_author" => [
    "Jeffrey Ryan Thalhammer <jeff\@imaginative-software.com>"
  ],
  "dist_name" => "Pinto-Server",
  "dist_version" => "0.50",
  "license" => "perl",
  "module_name" => "Pinto::Server",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Authen::Simple::Passwd" => 0,
    "Carp" => 0,
    "Class::Load" => 0,
    "File::Copy" => 0,
    "File::Temp" => 0,
    "File::Which" => 0,
    "Getopt::Long" => 0,
    "IO::Handle::Util" => 0,
    "IO::Interactive" => 0,
    "IO::Pipe" => 0,
    "IPC::Run" => 0,
    "JSON" => 0,
    "List::MoreUtils" => 0,
    "Log::Dispatch::Handle" => 0,
    "Moose" => 0,
    "MooseX::ClassAttribute" => 0,
    "MooseX::Types::Moose" => 0,
    "POSIX" => 0,
    "Path::Class" => 0,
    "Pinto" => "0.052",
    "Pinto::Constants" => 0,
    "Pinto::Result" => 0,
    "Pinto::Tester" => 0,
    "Pinto::Types" => 0,
    "Plack::MIME" => 0,
    "Plack::Middleware::Auth::Basic" => 0,
    "Plack::Request" => 0,
    "Plack::Response" => 0,
    "Plack::Runner" => 0,
    "Pod::Usage" => 0,
    "Proc::Fork" => 0,
    "Router::Simple" => 0,
    "Scalar::Util" => 0,
    "Test::TCP" => 0,
    "Try::Tiny" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [
    "bin/pintod"
  ]
);


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
